bin/ch/WScriptJsrt.cpp

```diff
@@ -960,25 +961,23 @@ bool WScriptJsrt::Uninitialize()
     if (GetRuntimeThreadLocalData().threadData && !GetRuntimeThreadLocalData().threadData->children.empty())
     {
         LONG count = (LONG)GetRuntimeThreadLocalData().threadData->children.size();
-        GetRuntimeThreadLocalData().threadData->hSemaphore = CreateSemaphore(NULL, 0, count, NULL);
+        std::vector<HANDLE> childrenHandles;
         
         //Clang does not support "for each" yet
         for(auto i = GetRuntimeThreadLocalData().threadData->children.begin(); i!= GetRuntimeThreadLocalData().threadData->children.end(); i++)
         {
             auto child = *i;
-            if (child->leaving)
-            {
-                ReleaseSemaphore(GetRuntimeThreadLocalData().threadData->hSemaphore, 1, NULL);
-            }
-            else
-            {
-                SetEvent(child->hevntShutdown);
-            }
+            childrenHandles.push_back(child->hThread);
+            SetEvent(child->hevntShutdown);
         }
 
-        WaitForSingleObject(GetRuntimeThreadLocalData().threadData->hSemaphore, INFINITE);
-        CloseHandle(GetRuntimeThreadLocalData().threadData->hSemaphore);
-        GetRuntimeThreadLocalData().threadData->hSemaphore = INVALID_HANDLE_VALUE;
+        DWORD waitRet = WaitForMultipleObjects(count, &childrenHandles[0], TRUE, INFINITE);
+        Assert(waitRet == WAIT_OBJECT_0);
+
+        for (auto i = GetRuntimeThreadLocalData().threadData->children.begin(); i != GetRuntimeThreadLocalData().threadData->children.end(); i++)
+        {
+            delete *i;
```

Comment:
I think @Penguinwizzard and ASAN would appreciate the cleanup even if it's technically unnecessary :)